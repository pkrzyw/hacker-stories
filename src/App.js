import React, {
  useState,
  useEffect,
  useRef,
  useReducer,
  useCallback,
} from "react";
import axios from "axios";
import "./App.css";
import { ReactComponent as Check } from "./check.svg";

const API_ENDPOINT = "https://hn.algolia.com/api/v1/search?query=";

const storiesReducer = (state, action) => {
  switch (action.type) {
    case "STORIES_FETCH_INIT":
      return { ...state, isLoading: true, isError: false };
    case "STORIES_FETCH_SUCCESS":
      return {
        ...state,
        isLoading: false,
        isError: false,
        data: action.payload,
      };
    case "STORIES_FETCH_FAILURE":
      return { ...state, isError: true, isLoading: false };
    case "REMOVE_STORY":
      return {
        ...state,
        data: state.data.filter(
          (story) => story.objectID !== action.payload.objectID
        ),
      };
    default:
      throw new Error();
  }
};

const useSemiPersistentState = (key, initialState) => {
  const [value, setValue] = useState(localStorage.getItem(key) || initialState);
  useEffect(() => {
    localStorage.setItem(key, value);
  }, [value, key]);

  return [value, setValue];
};

const App = () => {
  const [searchTerm, setSearchTerm] = useSemiPersistentState("search", "React");
  const [stories, dispatchStories] = useReducer(storiesReducer, {
    data: [],
    isLoading: false,
    isError: false,
  });
  const [url, setUrl] = useState(`${API_ENDPOINT}${searchTerm}`);

  const handleFetchStories = useCallback(async () => {
    dispatchStories({ type: "STORIES_FETCH_INIT" });
    try {
      const result = await axios.get(url);
      dispatchStories({
        type: "STORIES_FETCH_SUCCESS",
        payload: result.data.hits,
      });
    } catch {
      dispatchStories({ type: "STORIES_FETCH_FAILURE" });
    }
  }, [url]);

  useEffect(() => {
    handleFetchStories();
  }, [handleFetchStories]);

  const handleRemoveStory = (item) => {
    dispatchStories({
      type: "REMOVE_STORY",
      payload: item,
    });
  };

  const handleSearchInput = (event) => {
    setSearchTerm(event.target.value);
  };

  const handleSearchSubmit = (event) => {
    setUrl(`${API_ENDPOINT}${searchTerm}`);
    event.preventDefault();
  };

  const sumComments = getSumComments(stories);
  return (
    <div className="container">
      <h1 className="headline-primary">
        My Hacker Stories with {sumComments} comments.
      </h1>
      <SearchForm
        onSearchSubmit={handleSearchSubmit}
        searchTerm={searchTerm}
        onSearchInput={handleSearchInput}
      />
      {stories.isError && <p>Something went wrong ...</p>}
      {stories.isLoading ? (
        <p>Loading...</p>
      ) : (
        <List list={stories.data} onRemoveItem={handleRemoveStory} />
      )}
    </div>
  );
};
const SearchForm = ({ onSearchSubmit, searchTerm, onSearchInput }) => (
  <form onSubmit={onSearchSubmit} className="search-form">
    <InputWithLabel
      id="search"
      label="Search"
      value={searchTerm}
      onInputChange={onSearchInput}
      isFocused
    >
      <strong>Search:</strong>
    </InputWithLabel>
    <button
      disabled={!searchTerm}
      type="submit"
      className="button button_large"
    >
      Submit
    </button>
  </form>
);
const InputWithLabel = ({
  id,
  type = "text",
  value,
  onInputChange,
  isFocused,
  children,
}) => {
  const inputRef = useRef();
  useEffect(() => {
    if (isFocused && inputRef.current) {
      inputRef.current.focus();
    }
  }, [isFocused]);
  return (
    <>
      <label htmlFor={id} className="label">
        {children}
      </label>
      &nbsp;
      <input
        ref={inputRef}
        id={id}
        type={type}
        value={value}
        onChange={onInputChange}
        className="input"
      />
    </>
  );
};
const List = ({ list, onRemoveItem }) =>
  list.map((item) => (
    <Item item={item} key={item.objectID} onRemoveItem={onRemoveItem} />
  ));

const Item = ({ item, onRemoveItem }) => (
  <div className="item">
    <span style={{ width: "40%" }}>
      <a href={item.url}>{item.title}</a>{" "}
    </span>
    <span style={{ width: "30%" }}>{item.author}</span>{" "}
    <span style={{ width: "10%" }}>{item.num_comments}</span>{" "}
    <span style={{ width: "10%" }}>{item.points}</span>{" "}
    <span style={{ width: "10%" }}>
      <button
        type="button"
        onClick={() => onRemoveItem(item)}
        className="button button_small"
      >
        <Check height="18px" width="18px" />
      </button>
    </span>
  </div>
);

const getSumComments = (stories) => {
  return stories.data.reduce((result, value) => result + value.num_comments, 0);
};

export default App;
