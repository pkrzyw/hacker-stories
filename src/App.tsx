import React, {
  useState,
  useEffect,
  useRef,
  useReducer,
  useCallback,
} from "react";
import axios from "axios";
import "./App.css";
import { ReactComponent as Check } from "./check.svg";
type Story = {
  objectID: string;
  url: string;
  title: string;
  author: string;
  num_comments: number;
  points: number;
};
type ItemProps = {
  item: Story;
  onRemoveItem: (item: Story) => void;
};
type Stories = Array<Story>;
type ListProps = {
  list: Stories;
  onRemoveItem: (item: Story) => void;
};
type StoriesState = {
  data: Stories;
  isLoading: boolean;
  isError: boolean;
};

interface StoriesFetchInitAction {
  type: "STORIES_FETCH_INIT";
}
interface StoriesFetchSuccessAction {
  type: "STORIES_FETCH_SUCCESS";
  payload: Stories;
}
interface StoriesFetchFailureAction {
  type: "STORIES_FETCH_FAILURE";
}
interface StoriesRemoveAction {
  type: "REMOVE_STORY";
  payload: Story;
}
type StoriesAction =
  | StoriesFetchInitAction
  | StoriesFetchSuccessAction
  | StoriesFetchFailureAction
  | StoriesRemoveAction;
type SearchFormProps = {
  searchTerm: string;
  onSearchInput: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onSearchSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
};
const API_ENDPOINT = "https://hn.algolia.com/api/v1/search?query=";

const storiesReducer = (state: StoriesState, action: StoriesAction) => {
  switch (action.type) {
    case "STORIES_FETCH_INIT":
      return { ...state, isLoading: true, isError: false };
    case "STORIES_FETCH_SUCCESS":
      return {
        ...state,
        isLoading: false,
        isError: false,
        data: action.payload,
      };
    case "STORIES_FETCH_FAILURE":
      return { ...state, isError: true, isLoading: false };
    case "REMOVE_STORY":
      return {
        ...state,
        data: state.data.filter(
          (story) => story.objectID !== action.payload.objectID
        ),
      };
    default:
      throw new Error();
  }
};

const useSemiPersistentState = (
  key: string,
  initialState: string
): [string, (newValue: string) => void] => {
  const [value, setValue] = useState(localStorage.getItem(key) || initialState);
  useEffect(() => {
    localStorage.setItem(key, value);
  }, [value, key]);

  return [value, setValue];
};

const App = () => {
  const [searchTerm, setSearchTerm] = useSemiPersistentState("search", "React");
  const [stories, dispatchStories] = useReducer(storiesReducer, {
    data: [],
    isLoading: false,
    isError: false,
  });
  const [url, setUrl] = useState(`${API_ENDPOINT}${searchTerm}`);

  const handleFetchStories = useCallback(async () => {
    dispatchStories({ type: "STORIES_FETCH_INIT" });
    try {
      const result = await axios.get(url);
      dispatchStories({
        type: "STORIES_FETCH_SUCCESS",
        payload: result.data.hits,
      });
    } catch {
      dispatchStories({ type: "STORIES_FETCH_FAILURE" });
    }
  }, [url]);

  useEffect(() => {
    handleFetchStories();
  }, [handleFetchStories]);

  const handleRemoveStory = (item: Story) => {
    dispatchStories({
      type: "REMOVE_STORY",
      payload: item,
    });
  };

  const handleSearchInput = (event: React.ChangeEvent<HTMLInputElement>) => {
    setSearchTerm(event.target.value);
  };

  const handleSearchSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    setUrl(`${API_ENDPOINT}${searchTerm}`);
    event.preventDefault();
  };

  const sumComments = getSumComments(stories);

  return (
    <div className="container">
      <h1 className="headline-primary">
        My Hacker Stories with {sumComments} comments.
      </h1>
      <SearchForm
        onSearchSubmit={handleSearchSubmit}
        searchTerm={searchTerm}
        onSearchInput={handleSearchInput}
      />
      {stories.isError && <p>Something went wrong ...</p>}
      {stories.isLoading ? (
        <p>Loading...</p>
      ) : (
        <List list={stories.data} onRemoveItem={handleRemoveStory} />
      )}
    </div>
  );
};
const SearchForm = ({
  onSearchSubmit,
  searchTerm,
  onSearchInput,
}: SearchFormProps) => (
  <form onSubmit={onSearchSubmit} className="search-form">
    <InputWithLabel
      id="search"
      value={searchTerm}
      onInputChange={onSearchInput}
      isFocused
    >
      <strong>Search:</strong>
    </InputWithLabel>
    <button
      disabled={!searchTerm}
      type="submit"
      className="button button_large"
    >
      Submit
    </button>
  </form>
);
type InputWithLabelProps = {
  id: string;
  value: string;
  type?: string;
  onInputChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  isFocused?: boolean;
  children: React.ReactNode;
};
const InputWithLabel = ({
  id,
  type = "text",
  value,
  onInputChange,
  isFocused,
  children,
}: InputWithLabelProps) => {
  const inputRef = React.useRef<HTMLInputElement>(null!);
  useEffect(() => {
    if (isFocused && inputRef.current) {
      inputRef.current.focus();
    }
  }, [isFocused]);
  return (
    <>
      <label htmlFor={id} className="label">
        {children}
      </label>
      &nbsp;
      <input
        ref={inputRef}
        id={id}
        type={type}
        value={value}
        onChange={onInputChange}
        className="input"
      />
    </>
  );
};
const List = ({ list, onRemoveItem }: ListProps) => (
  <>
    {list.map((item) => (
      <Item item={item} key={item.objectID} onRemoveItem={onRemoveItem} />
    ))}
  </>
);

const Item = ({ item, onRemoveItem }: ItemProps) => (
  <div className="item">
    <span style={{ width: "40%" }}>
      <a href={item.url}>{item.title}</a>{" "}
    </span>
    <span style={{ width: "30%" }}>{item.author}</span>{" "}
    <span style={{ width: "10%" }}>{item.num_comments}</span>{" "}
    <span style={{ width: "10%" }}>{item.points}</span>{" "}
    <span style={{ width: "10%" }}>
      <button
        type="button"
        onClick={() => onRemoveItem(item)}
        className="button button_small"
      >
        <Check height="18px" width="18px" />
      </button>
    </span>
  </div>
);

const getSumComments = (stories: StoriesState) => {
  return stories.data.reduce((result, value) => result + value.num_comments, 0);
};

export default App;
