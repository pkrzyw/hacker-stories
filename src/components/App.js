import React, { useState, useEffect, useReducer, useCallback } from "react";
import axios from "axios";
import "./App.css";
import List from "./List";
import SearchForm from "./SearchForm";
import LastSearches from "./LastSearches";
const API_BASE = "https://hn.algolia.com/api/v1";
const API_SEARCH = "/search";
const PARAM_SEARCH = "query=";
const PARAM_PAGE = "page=";

const getUrl = (searchTerm, page) =>
  `${API_BASE}${API_SEARCH}?${PARAM_SEARCH}${searchTerm}&${PARAM_PAGE}${page}`;
export const storiesReducer = (state, action) => {
  switch (action.type) {
    case "STORIES_FETCH_INIT":
      return { ...state, isLoading: true, isError: false };
    case "STORIES_FETCH_SUCCESS":
      return {
        ...state,
        isLoading: false,
        isError: false,
        data:
          action.payload.page === 0
            ? action.payload.list
            : state.data.concat(action.payload.list),
        page: action.payload.page,
      };
    case "STORIES_FETCH_FAILURE":
      return { ...state, isError: true, isLoading: false };
    case "REMOVE_STORY":
      return {
        ...state,
        data: state.data.filter(
          (story) => story.objectID !== action.payload.objectID
        ),
      };
    default:
      throw new Error();
  }
};

const useSemiPersistentState = (key, initialState) => {
  const [value, setValue] = useState(localStorage.getItem(key) || initialState);
  useEffect(() => {
    localStorage.setItem(key, value);
  }, [value, key]);

  return [value, setValue];
};

const extractSearchTerm = (url) =>
  url
    .substring(url.lastIndexOf("?") + 1, url.lastIndexOf("&"))
    .replace(PARAM_SEARCH, "");
const getLastSearches = (urls) =>
  urls
    .reduce((result, url, index) => {
      const searchTerm = extractSearchTerm(url);
      if (index === 0) {
        return result.concat(searchTerm);
      }
      const previousSearchTerm = result[result.length - 1];
      if (searchTerm === previousSearchTerm) {
        return result;
      } else {
        return result.concat(searchTerm);
      }
    }, [])
    .slice(-6)
    .slice(0, -1);

const App = () => {
  const [searchTerm, setSearchTerm] = useSemiPersistentState("search", "React");
  const [urls, setUrls] = useState([getUrl(searchTerm, 0)]);
  const [stories, dispatchStories] = useReducer(storiesReducer, {
    data: [],
    page: 0,
    isLoading: false,
    isError: false,
  });

  const handleFetchStories = useCallback(async () => {
    dispatchStories({ type: "STORIES_FETCH_INIT" });
    try {
      const result = await axios.get(urls[urls.length - 1]);
      console.log(result);
      dispatchStories({
        type: "STORIES_FETCH_SUCCESS",
        payload: {
          list: result.data.hits,
          page: result.data.page,
        },
      });
    } catch {
      dispatchStories({ type: "STORIES_FETCH_FAILURE" });
    }
  }, [urls]);

  useEffect(() => {
    handleFetchStories();
  }, [handleFetchStories]);
  const handleRemoveStory = (item) => {
    dispatchStories({
      type: "REMOVE_STORY",
      payload: item,
    });
  };

  const handleSearchInput = (event) => {
    setSearchTerm(event.target.value);
  };

  const handleSearchSubmit = (event) => {
    handleSearch(searchTerm, 0);
    event.preventDefault();
  };
  const handleLastSearch = (searchTerm) => {
    setSearchTerm(searchTerm);
    handleSearch(searchTerm, 0);
  };

  const handleSearch = (searchTerm, page) => {
    const url = getUrl(searchTerm, page);
    setUrls(urls.concat(url));
  };
  const handleMore = () => {
    const lastUrl = urls[urls.length - 1];
    const searchTerm = extractSearchTerm(lastUrl);
    handleSearch(searchTerm, stories.page + 1);
  };

  const sumComments = getSumComments(stories);

  const lastSearches = getLastSearches(urls);
  return (
    <div className="container">
      <h1 className="headline-primary">
        My Hacker Stories with {sumComments} comments.
      </h1>
      <SearchForm
        onSearchSubmit={handleSearchSubmit}
        searchTerm={searchTerm}
        onSearchInput={handleSearchInput}
      />

      <LastSearches
        lastSearches={lastSearches}
        onLastSearch={handleLastSearch}
      />
      <hr />
      {stories.isError && <p>Something went wrong ...</p>}
      <List list={stories.data} onRemoveItem={handleRemoveStory} />
      {stories.isLoading ? (
        <p>Loading...</p>
      ) : (
        <button
          type="button"
          onClick={handleMore}
          className="button button_large"
        >
          {" "}
          More
        </button>
      )}
    </div>
  );
};

const getSumComments = (stories) => {
  return stories.data.reduce((result, value) => result + value.num_comments, 0);
};

export default App;
