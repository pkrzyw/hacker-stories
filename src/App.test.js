import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import App, {
  Item,
  List,
  SearchForm,
  InputWithLabel,
  storiesReducer,
} from "./App";
import axios from "axios";
jest.mock("axios");

describe("Item", () => {
  const item = {
    title: "React",
    url: "https://reactjs.org/",
    author: "Jordan Walke",
    num_comments: 3,
    points: 4,
    objectID: 0,
  };
  const handleRemoveItem = jest.fn();
  let component;

  beforeEach(() => {
    component = renderer.create(
      <Item item={item} onRemoveItem={handleRemoveItem} />
    );
  });

  it("renders all properties", () => {
    expect(component.root.findByType("a").props.href).toEqual(
      "https://reactjs.org/"
    );
    expect(
      component.root.findAllByProps({ children: "Jordan Walke" }).length
    ).toEqual(1);
    expect(component.root.findAllByProps({ children: "React" }).length).toEqual(
      1
    );
  });
  it("calls onRemoveItem on button click", () => {
    component.root.findByType("button").props.onClick();

    expect(handleRemoveItem).toHaveBeenCalledTimes(1);
    expect(handleRemoveItem).toHaveBeenCalledWith(item);

    expect(component.root.findAllByType(Item).length).toEqual(1);
  });
  it("renders snapshot", () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe("List", () => {
  const list = [
    {
      title: "React",
      url: "https://reactjs.org/",
      author: "Jordan Walke",
      num_comments: 3,
      points: 4,
      objectID: 0,
    },
    {
      title: "Redux",
      url: "https://redux.js.org/",
      author: "Dan Abramov, Andrew Clark",
      num_comments: 2,
      points: 5,
      objectID: 1,
    },
  ];
  it("renders two items", () => {
    const component = renderer.create(<List list={list} />);
    expect(component.root.findAllByType(Item).length).toEqual(2);
  });
  test("renders snapshot", () => {
    const component = renderer.create(<List list={list} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe("SearchForm", () => {
  const searchFormProps = {
    searchTerm: "React",
    onSearchInput: jest.fn(),
    onSearchSubmit: jest.fn(),
  };
  let component;
  beforeEach(() => {
    component = renderer.create(<SearchForm {...searchFormProps} />);
  });
  it("renders the input field with its value", () => {
    const value = component.root.findByType("input").props.value;
    expect(value).toEqual("React");
  });
  it("changes the input field", () => {
    const pseudoEvent = { target: "Redux" };
    component.root.findByType("input").props.onChange(pseudoEvent);
    expect(searchFormProps.onSearchInput).toHaveBeenCalledTimes(1);
    expect(searchFormProps.onSearchInput).toHaveBeenCalledWith(pseudoEvent);
  });
  it("submits the form", () => {
    const pseudoEvent = {};
    component.root.findByType("form").props.onSubmit(pseudoEvent);
    expect(searchFormProps.onSearchSubmit).toHaveBeenCalledTimes(1);
    expect(searchFormProps.onSearchSubmit).toHaveBeenCalledWith(pseudoEvent);
  });
  it("disables the button and prevents submit", () => {
    component.update(<SearchForm {...searchFormProps} searchTerm="" />);
    expect(component.root.findByType("button").props.disabled).toBeTruthy();
  });
  test("renders snapshot", () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe("Separate Enzyme tests", () => {
  it("renders child InputWithLabel", () => {
    const wrapper = mount(<SearchForm />);
    expect(wrapper.find(InputWithLabel).length).toEqual(1);
  });
  it("renders child InputWithLabel with 'Search:' label", () => {
    const wrapper = mount(<SearchForm />);
    expect(wrapper.find("strong").text()).toEqual("Search:");
  });
});
describe("App", () => {
  it("renders child InputWithLabel and responds to new values", () => {
    const wrapper = mount(<App />);
    wrapper
      .find("input")
      .at(0)
      .simulate("change", { target: { value: "ReasonML" } });
    const InputWrapper = wrapper.find(InputWithLabel);
    expect(InputWrapper.find("input").prop("value")).toEqual("ReasonML");
  });
  it("succeeds fetching data with a list", async () => {
    const list = [
      {
        title: "React",
        url: "https://reactjs.org/",
        author: "Jordan Walke",
        num_comments: 3,
        points: 4,
        objectID: 0,
      },
      {
        title: "Redux",
        url: "https://redux.js.org/",
        author: "Dan Abramov, Andrew Clark",
        num_comments: 2,
        points: 5,
        objectID: 1,
      },
    ];
    const promise = Promise.resolve({
      data: {
        hits: list,
      },
    });
    axios.get.mockImplementationOnce(() => promise);
    let component;
    await renderer.act(async () => {
      component = renderer.create(<App />);
    });
    expect(component.root.findByType(List).props.list).toEqual(list);
  });
  it("fails fetching data with a list", async () => {
    const promise = Promise.reject();
    axios.get.mockImplementationOnce(() => promise);
    let component;
    await renderer.act(async () => {
      component = renderer.create(<App />);
    });
    expect(component.root.findByType("p").props.children).toEqual(
      "Something went wrong ..."
    );
  });
  it("renders snapshot", () => {
    let component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe("storiesReducer", () => {
    const state = {
      data: [
        {
          title: "React",
          url: "https://reactjs.org/",
          author: "Jordan Walke",
          num_comments: 3,
          points: 4,
          objectID: 0,
        },
        {
          title: "Redux",
          url: "https://redux.js.org/",
          author: "Dan Abramov, Andrew Clark",
          num_comments: 2,
          points: 5,
          objectID: 1,
        },
      ],
      isLoading: false,
      isError: false,
    };
    it("should set fetch init", () => {
      const newState = storiesReducer(state, { type: "STORIES_FETCH_INIT" });
      expect(newState).toEqual({ ...state, isLoading: true, isError: false });
    });
    it("should set fetch error", () => {
      const newState = storiesReducer(state, { type: "STORIES_FETCH_FAILURE" });
      expect(newState).toEqual({ ...state, isLoading: false, isError: true });
    });
    it("should set fetch success", () => {
      const newState = storiesReducer(state, {
        type: "STORIES_FETCH_SUCCESS",
        payload: [
          {
            title: "React",
            url: "https://reactjs.org/",
            author: "Jordan Walke",
            num_comments: 3,
            points: 5,
            objectID: 0,
          },
          {
            title: "Redux",
            url: "https://redux.js.org/",
            author: "Dan Abramov, Andrew Clark",
            num_comments: 2,
            points: 6,
            objectID: 1,
          },
        ],
      });
      expect(newState).toEqual({
        data: [
          {
            title: "React",
            url: "https://reactjs.org/",
            author: "Jordan Walke",
            num_comments: 3,
            points: 5,
            objectID: 0,
          },
          {
            title: "Redux",
            url: "https://redux.js.org/",
            author: "Dan Abramov, Andrew Clark",
            num_comments: 2,
            points: 6,
            objectID: 1,
          },
        ],
        isLoading: false,
        isError: false,
      });
    });
    it("should remove a story", () => {
      const newState = storiesReducer(state, {
        type: "REMOVE_STORY",
        payload: {
          title: "React",
          url: "https://reactjs.org/",
          author: "Jordan Walke",
          num_comments: 3,
          points: 4,
          objectID: 0,
        },
      });
      expect(newState).toEqual({
        data: [
          {
            title: "Redux",
            url: "https://redux.js.org/",
            author: "Dan Abramov, Andrew Clark",
            num_comments: 2,
            points: 5,
            objectID: 1,
          },
        ],
        isLoading: false,
        isError: false,
      });
    });
  });
});
