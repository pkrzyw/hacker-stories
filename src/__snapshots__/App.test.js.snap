// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders snapshot 1`] = `
<div
  className="container"
>
  <h1
    className="headline-primary"
  >
    My Hacker Stories with 
    0
     comments.
  </h1>
  <form
    className="search-form"
    onSubmit={[Function]}
  >
    <label
      className="label"
      htmlFor="search"
    >
      <strong>
        Search:
      </strong>
    </label>
     
    <input
      className="input"
      id="search"
      onChange={[Function]}
      type="text"
      value="ReasonML"
    />
    <button
      className="button button_large"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`Item renders snapshot 1`] = `
<div
  className="item"
>
  <span
    style={
      Object {
        "width": "40%",
      }
    }
  >
    <a
      href="https://reactjs.org/"
    >
      React
    </a>
     
  </span>
  <span
    style={
      Object {
        "width": "30%",
      }
    }
  >
    Jordan Walke
  </span>
   
  <span
    style={
      Object {
        "width": "10%",
      }
    }
  >
    3
  </span>
   
  <span
    style={
      Object {
        "width": "10%",
      }
    }
  >
    4
  </span>
   
  <span
    style={
      Object {
        "width": "10%",
      }
    }
  >
    <button
      className="button button_small"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="18px"
        width="18px"
      >
        check.svg
      </svg>
    </button>
  </span>
</div>
`;

exports[`List renders snapshot 1`] = `
Array [
  <div
    className="item"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href="https://reactjs.org/"
      >
        React
      </a>
       
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      Jordan Walke
    </span>
     
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      3
    </span>
     
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      4
    </span>
     
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button button_small"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="18px"
          width="18px"
        >
          check.svg
        </svg>
      </button>
    </span>
  </div>,
  <div
    className="item"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href="https://redux.js.org/"
      >
        Redux
      </a>
       
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      Dan Abramov, Andrew Clark
    </span>
     
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      2
    </span>
     
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      5
    </span>
     
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button button_small"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="18px"
          width="18px"
        >
          check.svg
        </svg>
      </button>
    </span>
  </div>,
]
`;

exports[`SearchForm renders snapshot 1`] = `
<form
  className="search-form"
  onSubmit={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {},
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <label
    className="label"
    htmlFor="search"
  >
    <strong>
      Search:
    </strong>
  </label>
   
  <input
    className="input"
    id="search"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "target": "Redux",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    type="text"
    value="React"
  />
  <button
    className="button button_large"
    disabled={false}
    type="submit"
  >
    Submit
  </button>
</form>
`;
